### PR TITLE
Update the python API docs site for the new APIs

### PR DESCRIPTION
**Pull request by jleibs**
### What
 - Builds on top of: https://github.com/rerun-io/rerun/pull/3564
 - Resolves: https://github.com/rerun-io/rerun/issues/3437
 
Replaces the logging functions with the archetypes.
Moves the deprecated function to their own section.

Also fixes the color inversion issue.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3565) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3565)
- [Docs preview](https://rerun.io/preview/85bdc0990d13a21c0cf1c2385e6601af2a9edaf8/docs) 
- [Examples preview](https://rerun.io/preview/85bdc0990d13a21c0cf1c2385e6601af2a9edaf8/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: rerun_py/docs/gen_common_index.py [hunk 1]
```diff
@@ -54,88 +54,58 @@ class Section:
...
-            "log_line_strip",
-            "log_line_segments",
-            "log_arrow",
-            "log_mesh",
-            "log_meshes",
-            "log_mesh_file",
-        ],
-        class_list=[],
+        func_list=[],
+        class_list=["Points2D", "Points3D", "Boxes2D", "Boxes3D", "LineStrips2D", "LineStrips3D", "Mesh3D", "Asset3D"],
```
Comment (Wumpf) [inline rerun_py/docs/gen_common_index.py:66]: Arrow is missing

File: rerun_py/docs/package/index.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # Full Package
 
 - [rerun][]: Top-level User-facing APIs. See also: [Common APIs](../common/)
-    - [rerun.log][]: APIs for logging data
-    - [rerun.color_conversion][]: Conversion utilities related to colors
-    - [rerun.components][]: Helpers for constructing arrow components
+    - [rerun.archetypes][]: Common objects for logging primitives understood by the viewer, e.g. Points and Lines.
```
Comment (Wumpf) [inline rerun_py/docs/package/index.md:4]:
```suggestion
    - [rerun.archetypes][]: Common objects for logging tagged data understood by the viewer, e.g. Points and Lines.
```

maybe better? I'm getting more and more uneasy around "primitives". Putting `tagged` here might be too much though (also we never call the indicators 'tagged'), but maybe something like this

**Review by Wumpf — APPROVED** [review]

File: rerun_py/docs/gen_common_index.py (hunk 1 above)
Comment (emilk) [reply rerun_py/docs/gen_common_index.py:66]: Also: what is this order? O.o

Comment (jleibs) [reply rerun_py/docs/gen_common_index.py:66]: Ordered by increasing complexity of object. Same as how the functions were ordered.

**Comment by abey79** [comment]
The generated documentation structure is pretty badly broken for me. This is clicking Point2d from the "landing page":

image

Notice no title or anything for the class. Also note TOC on the right hand side.

On Safari, this is aggravated by layout issues:

image

**Pull request merged**